src/algebra/free_algebra.lean

```diff
@@ -137,7 +144,7 @@ local attribute [instance]
   pre.has_coe_generator pre.has_coe_semiring pre.has_mul pre.has_add pre.has_zero
   pre.has_one pre.has_scalar
 
-instance : semiring (free_algebra R X) :=
+instance : add_comm_monoid (free_algebra R X) :=
```

Comment:
What's the reason to split instances?

Reply:
In a first version of the PR, the `nat_cast` field did not have a default value.  Splitting the instance was necessary to include the `add_monoid_with_one.unary` definition (which has an `[add_monoid R]` argument).

There is a default value now, so I've reverted all changes to this file.